### PR TITLE
[Changes Requested From Developer] Optimizations

### PR DESCRIPTION
**Pull request by jondionowens**
Opted to not use models for insertion, though I maintained an option to go back and forth if needed.
Also added simple console progress tracking.

### CODE REVIEW

**Comment by MengRS** [comment]
@jondionowens,

I'm getting the following when I query the mongodb:
```
        "_id" : ObjectId("5c4fffe45b6ae9053a07a802"),
        "listing_id" : 1,
        "details" : [
                {
                        "date" : {
                                "_isAMomentObject" : true,
                                "_i" : ISODate("2018-07-06T17:31:01.575Z"),
                                "_isUTC" : false,
                                "_pf" : {
                                        "empty" : false,
                                        "unusedTokens" : [ ],
                                        "unusedInput" : [ ],
                                        "overflow" : -2,
                                        "charsLeftOver" : 0,
                                        "nullInput" : false,
                                        "invalidMonth" : null,
                                        "invalidFormat" : false,
                                        "userInvalidated" : false,
                                        "iso" : false,
                                        "parsedDateParts" : [ ],
                                        "meridiem" : null,
                                        "rfc2822" : false,
                                        "weekdayMismatch" : false
                                },
                                "_locale" : {
                                        "_calendar" : {
                                                "sameDay" : "[Today at] LT",
                                                "nextDay" : "[Tomorrow at] LT",
                                                "nextWeek" : "dddd [at] LT",
                                                "lastDay" : "[Yesterday at] LT",
                                                "lastWeek" : "[Last] dddd [at] LT",
                                                "sameElse" : "L"
                                        },
                                        "_longDateFormat" : {
                                                "LTS" : "h:mm:ss A",
                                                "LT" : "h:mm A",
                                                "L" : "MM/DD/YYYY",
                                                "LL" : "MMMM D, YYYY",
                                                "LLL" : "MMMM D, YYYY h:mm A",
                                                "LLLL" : "dddd, MMMM D, YYYY h:mm A"
                                        },
                                        "_invalidDate" : "Invalid date",
                                        "_dayOfMonthOrdinalParse" : /\d{1,2}(th|st|nd|rd)/,
                                        "_relativeTime" : {
                                                "future" : "in %s",
                                                "past" : "%s ago",
                                                "s" : "a few seconds",
                                                "ss" : "%d seconds",
                                                "m" : "a minute",
                                                "mm" : "%d minutes",
                                                "h" : "an hour",
                                                "hh" : "%d hours",
                                                "d" : "a day",
                                                "dd" : "%d days",
                                                "M" : "a month",
                                                "MM" : "%d months",
                                                "y" : "a year",
                                                "yy" : "%d years"
                                        },
                                        "_months" : [
                                                "January",
                                                "February",
                                                "March",
                                                "April",
                                                "May",
                                                "June",
                                                "July",
                                                "August",
                                                "September",
                                                "October",
                                                "November",
                                                "December"
                                        ],
                                        "_monthsShort" : [
                                                "Jan",
                                                "Feb",
                                                "Mar",
                                                "Apr",
                                                "May",
                                                "Jun",
                                                "Jul",
                                                "Aug",
                                                "Sep",
                                                "Oct",
                                                "Nov",
                                                "Dec"
                                        ],
                                        "_week" : {
                                                "dow" : 0,
                                                "doy" : 6
                                        },
                                        "_weekdays" : [
                                                "Sunday",
                                                "Monday",
                                                "Tuesday",
                                                "Wednesday",
                                                "Thursday",
                                                "Friday",
                                                "Saturday"
                                        ],
                                        "_weekdaysMin" : [
                                                "Su",
                                                "Mo",
                                                "Tu",
                                                "We",
                                                "Th",
                                                "Fr",
                                                "Sa"
                                        ],
                                        "_weekdaysShort" : [
                                                "Sun",
                                                "Mon",
                                                "Tue",
                                                "Wed",
                                                "Thu",
                                                "Fri",
                                                "Sat"
                                        ],
                                        "_meridiemParse" : /[ap]\.?m?\.?/i,
                                        "_abbr" : "en",
                                        "_config" : {
                                                "calendar" : {
                                                        "sameDay" : "[Today at] LT",
                                                        "nextDay" : "[Tomorrow at] LT",
                                                        "nextWeek" : "dddd [at] LT",
                                                        "lastDay" : "[Yesterday at] LT",
                                                        "lastWeek" : "[Last] dddd [at] LT",
                                                        "sameElse" : "L"
                                                },
                                                "longDateFormat" : {
                                                        "LTS" : "h:mm:ss A",
                                                        "LT" : "h:mm A",
                                                        "L" : "MM/DD/YYYY",
                                                        "LL" : "MMMM D, YYYY",
                                                        "LLL" : "MMMM D, YYYY h:mm A",
                                                        "LLLL" : "dddd, MMMM D, YYYY h:mm A"
                                                },
                                                "invalidDate" : "Invalid date",
                                                "dayOfMonthOrdinalParse" : /\d{1,2}(th|st|nd|rd)/,
                                                "relativeTime" : {
                                                        "future" : "in %s",
                                                        "past" : "%s ago",
                                                        "s" : "a few seconds",
                                                        "ss" : "%d seconds",
                                                        "m" : "a minute",
                                                        "mm" : "%d minutes",
                                                        "h" : "an hour",
                                                        "hh" : "%d hours",
                                                        "d" : "a day",
                                                        "dd" : "%d days",
                                                        "M" : "a month",
                                                        "MM" : "%d months",
                                                        "y" : "a year",
                                                        "yy" : "%d years"
                                                },
                                                "months" : [
                                                        "January",
                                                        "February",
                                                        "March",
                                                        "April",
                                                        "May",
                                                        "June",
                                                        "July",
                                                        "August",
                                                        "September",
                                                        "October",
                                                        "November",
                                                        "December"
                                                ],
                                                "monthsShort" : [
                                                        "Jan",
                                                        "Feb",
                                                        "Mar",
                                                        "Apr",
                                                        "May",
                                                        "Jun",
                                                        "Jul",
                                                        "Aug",
                                                        "Sep",
                                                        "Oct",
                                                        "Nov",
                                                        "Dec"
                                                ],
                                                "week" : {
                                                        "dow" : 0,
                                                        "doy" : 6
                                                },
                                                "weekdays" : [
                                                        "Sunday",
                                                        "Monday",
                                                        "Tuesday",
                                                        "Wednesday",
                                                        "Thursday",
                                                        "Friday",
                                                        "Saturday"
                                                ],
                                                "weekdaysMin" : [
                                                        "Su",
                                                        "Mo",
                                                        "Tu",
                                                        "We",
                                                        "Th",
                                                        "Fr",
                                                        "Sa"
                                                ],
                                                "weekdaysShort" : [
                                                        "Sun",
                                                        "Mon",
                                                        "Tue",
                                                        "Wed",
                                                        "Thu",
                                                        "Fri",
                                                        "Sat"
                                                ],
                                                "meridiemParse" : /[ap]\.?m?\.?/i,
                                                "abbr" : "en"
                                        },
                                        "_dayOfMonthOrdinalParseLenient" : /\d{1,2}(th|st|nd|rd)|\d{1,2}/
                                },
                                "_d" : ISODate("2018-07-06T07:00:00Z"),
                                "_isValid" : true
                        },
                        "guests" : {
                                "adults" : 1,
                                "children" : 0,
                                "infants" : 3
                        }
                }
        ],
        "listing_price" : "64.00"
}

```

In addition, when I run `npm start` and try to ping the first listing_id, I get "No Listing Found", after I seeded and verified there is a `listing_id`: 1 present.

File: server/seeder.js [hunk 1]
```diff
@@ -1,60 +1,122 @@
...
+    batchCounter++;
+  }
+
+  if (progressInserted === (recordsPerBatch * batches)) {
+    console.log('Done!');
+  }
+};
+
+const insert = (useModel) => {
+  if (useModel) {
```
Comment (MengRS) [inline server/seeder.js]: LOVE this!!!!

File: server/seeder.js [hunk 2]
```diff
@@ -1,60 +1,122 @@
...
-      }) 
-
-     newListing.save((err) => {
-        if(err){
-          console.log(err)
+      });
+    });
+  } else {
+    return new Promise(function (resolve, reject) {
+      db.collection('listing').insertMany(insertData, function (error, doc) {
```
Comment (MengRS) [inline server/seeder.js]: I believe if you change from `db.collection('listing').insertMany(insertData, function (error, doc) {` to `db.collection('listings').insertMany(insertData, function (error, doc) {` - your api will work.

File: server/seeder.js [hunk 3]
```diff
@@ -1,60 +1,122 @@
...
+  let recordsCounter = 1;
+  let batch = [];
+
+  while (recordsCounter <= recordsPerBatch) {
+    let details = [];
+    let bookingsCounter = 1;
+
+    while (bookingsCounter <= 1) {
+      let d = faker.date.between('2018-01-01', '2019-09-30');
+      const newD = moment(d).startOf('day');
```
Comment (MengRS) [inline server/seeder.js]: Delete this line

File: server/seeder.js [hunk 4]
```diff
@@ -1,60 +1,122 @@
...
+      detail = {
+        date: newD,
```
Comment (MengRS) [inline server/seeder.js]: replace `newD` with `d` and the I believe the date will render properly

**Review by MengRS — CHANGES_REQUESTED** [review]
Hi @jondionowens,

- In addition to the suggested changes proposed - I believe the over all data shape should end up looking like [this](https://github.com/rpt09-orchid/jon-booking#Data) when you hit the [api](http://localhost:3004/booking/1).  Even if you implement the suggested changes, it looks like only two fields are showing up:


- Lastly, I noticed the `listing_id` generated is always `1`. (See further suggestions on the commit code - where I think you just have to change a couple of things in certain spots to get it to count correctly).

Btw - let me know if you already knew that these things were broken - that's cool too and I can merge. :D

Btw - loved your seeder code where you could flip back and forth on whether or not we use the model! Good job!

**Comment by jondionowens** [comment]
Hey @MengRS. Thanks for this review. Yeah, lots of slop in here that I'm going to fix. Will get a update up as soon as possible. I plan on pulling a late night tonight

File: server/seeder.js (hunk 2 above)
Comment (jondionowens) [reply server/seeder.js]: derp. thanks

File: server/seeder.js (hunk 3 above)
Comment (jondionowens) [reply server/seeder.js]: Thanks!

File: server/seeder.js (hunk 4 above)
Comment (jondionowens) [reply server/seeder.js]: done

**Comment by jondionowens** [comment]
Hey @MengRS , updated pushed up here for your review. Thanks for the eyes!

**Comment by MengRS** [comment]
- 👍🏼 on my machine, the seeder ran in 5.43 minutes. 👍🏼


- 👍🏼 validated listing_id incrementation 👍🏼

**Comment by MengRS** [comment]
@jondionowens, when I ping your api - I still get:


I could be wrong, but I believe hitting the api - should yield a data shape more like [this](https://github.com/rpt09-orchid/jon-booking#Data)?  Let me know if you haven't gotten there yet / not going to focus on that / that was not the intention of the pull request and I can just merge it.  

Everything else looks great!

**Comment by jondionowens** [comment]
Hey there @MengRS. Sorry for the massive delay in getting back to this. I've been slacking in general ;)
Um, a few things)
I think the documentation about the same of the data that should be returned by the API you specified [here](https://github.com/rpt09-orchid/jon-booking#Data) might be out of date. I don't know what other services might have been referring to this endpoint, but the code definitely is purposely returning a different shape. I'm a little reluctant to change it until I know who else is using this data.

Pull down the latest and try to hit [http://localhost:3004/booking/1](http://localhost:3004/booking/1) Things should work as expected now. But before you do, make sure to re-run the seeder to generate some data in the correct shape. Maybe don't wait for 10 mill, just change the vars to provide a couple records.

Anywho, things seem to be working correctly now.

**Comment by MengRS** [comment]
Hi @jondionowens! I hear you on the doc not depicting the most up to date data shape - thus - can you tell me what data shape I should be expecting? I re-seeded and I'm still experiencing this one:
image

If the above is the expected outcome, then I'll go ahead and merge it - let me know.

**Comment by MengRS** [comment]
spoke to @jondionowens via slack - we're all good! Merging now!

**Pull request merged**